lib/services/database/post_upvote_service.dart

```diff
@@ -0,0 +1,131 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_data.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/upvote_state.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/services/database/firestore_service.dart";
+
+/// This repository service is responsible for handling the upvotes of posts
+class PostUpvoteRepositoryService {
+  /// The name of the subcollection that contains the list of users who upvoted
+  static const upvotersSubCollectionName = "upvoters";
+
+  /// The name of the subcollection that contains the list of users who downvoted
+  static const downvotersSubCollectionName = "downvoters";
+
+  final FirebaseFirestore firestore;
```

Comment:
I think this field should be private. Take a look at other Database related services like `UserRepositoryService` for the constructor.

Reply:
Definitely indeed, well seen! :)
Fixed in c65785d466d56d027069644486317e04ced3f76f.